src/test/ensemble_flow_custom_node_tests.cpp

```diff
@@ -0,0 +1,974 @@
+    const float subValue = 7.3;
+
+    std::vector<NodeInfo> info{
+        {NodeKind::ENTRY, ENTRY_NODE_NAME, "", std::nullopt, {{pipelineInputName, pipelineInputName}}},
+        {NodeKind::CUSTOM, "custom_node", "", std::nullopt, {{customNodeOutputName, customNodeOutputName}},
+            std::nullopt, std::nullopt, library, parameters_t{{"add_value", std::to_string(addValue)}, {"sub_value", std::to_string(subValue)}}},
+        {NodeKind::EXIT, EXIT_NODE_NAME},
+    };
+
+    pipeline_connections_t connections;
+
+    // request (pipelineInputName) O--------->O custom node (customNodeInputName)
+    connections["custom_node"] = {
+        {ENTRY_NODE_NAME, {{pipelineInputName, customNodeInputName}}}};
+
+    // custom node (customNodeOutputName) O--------->O response (pipelineOutputName)
+    connections[EXIT_NODE_NAME] = {
+        {"custom_node", {{customNodeOutputName, pipelineOutputName}}}};
+
+    std::unique_ptr<Pipeline> pipeline;
+    ASSERT_EQ(factory.createDefinition("my_new_pipeline", info, connections, manager), StatusCode::OK);
+    ASSERT_EQ(factory.create(pipeline, "my_new_pipeline", &request, &response, manager), StatusCode::OK);
+    ASSERT_EQ(pipeline->execute(), StatusCode::OK);
+
+    this->checkResponse<float>(inputValues, [addValue, subValue](float value) -> float {
+        return value + addValue - subValue;
+    });
+}
+
+TEST_F(EnsembleFlowCustomNodeFactoryCreateThenExecuteTest, ParallelPipelineFactoryUsageWithCustomNode) {
+    //                 Nodes
+    //              custom_node_N
+    //         v-------->O----------v
+    //  request O--------->O---------->O response     x   PARALLEL_SIMULATED_REQUEST_COUNT
+    //         ^-------->O----------^
+    //                add-sub
+    ConstructorEnabledModelManager manager;
+    PipelineFactory factory;
+
+    const int PARALLEL_CUSTOM_NODES = 3;
+    const int PARALLEL_SIMULATED_REQUEST_COUNT = 30;
+
+    const std::vector<float> inputValues{7.8, -2.4, 1.9, 8.7, -2.4, 3.5};
+    std::array<PredictRequest, PARALLEL_SIMULATED_REQUEST_COUNT> requests{};
+
+    for (int i = 0; i < PARALLEL_SIMULATED_REQUEST_COUNT; i++) {
+        this->prepareRequest(requests[i], inputValues);
+    }
+
+    const std::array<float, PARALLEL_CUSTOM_NODES> addValue{-1.5, 1.4, -0.1};
+    const std::array<float, PARALLEL_CUSTOM_NODES> subValue{4.9, -1.9, -0.9};
+
+    std::vector<NodeInfo> info{
+        {NodeKind::ENTRY, ENTRY_NODE_NAME, "", std::nullopt, {{pipelineInputName, pipelineInputName}}},
+        {NodeKind::EXIT, EXIT_NODE_NAME},
+    };
+
+    for (int i = 0; i < PARALLEL_CUSTOM_NODES; i++) {
+        info.emplace_back(std::move(NodeInfo(
+            NodeKind::CUSTOM,
+            "custom_node_" + std::to_string(i),
+            "", std::nullopt,
+            {{customNodeOutputName, customNodeOutputName}},
+            std::nullopt, std::nullopt,
+            library, parameters_t{{"add_value", std::to_string(addValue[i])}, {"sub_value", std::to_string(subValue[i])}})));
```

Comment:
Minor:
Maybe move L578 where you declare addValue, subValue before definition of PARALLEL_CUSTOM_NODES so that PARALLEL_CUSTOM_NODES = addValue.size()?

Minor:
addValue->addValues and the same with subValue since those are containers.

Reply:
Changed to plural, but do not see value in changing PARALLEL_CUSTOM_NODES to addValue.size()